Framework/Kernel/inc/MantidKernel/MagneticIon.h

```diff
@@ -20,9 +20,8 @@ struct MANTID_KERNEL_DLL MagneticIon {
   /// Default constructor
   MagneticIon();
   /// Construct the Ion with data
-  MagneticIon(const std::string symbol, const uint16_t charge,
-              const double j0[8], const double j2[8], const double j4[8],
-              const double j6[8]);
+  MagneticIon(const char *symbol, const uint16_t charge, const double j0[8],
+              const double j2[8], const double j4[8], const double j6[8]);
```

Comment:
Maybe the whole class could be made `constexpr`, if we manage to get rid of the `std::string`?